libtransmission/peer-mgr.cc

```diff
@@ -2450,10 +2450,11 @@ struct peer_candidate
         candidates.resize(max);
     }
 
+    // put the best candiates at the end of the list
     auto ret = tr_peerMgr::OutboundCandidates{};
-    for (auto const& candidate : candidates)
+    for (auto it = std::crbegin(candidates); it != std::crend(candidates); ++it)
```

Comment:
I usually avoid calls to `end()` inside a loop like this on the theory that they have _some_ minor cost. TBH I haven't actually investigated this so the premise may be faulty, so if you know better please LMK :smile_cat: 

Anyway, assuming that `std::crend()` isn't free:

```suggestion
    for (auto it = std::crbegin(candidates), end = std::crend(candidates); it != end; ++it)
```